### PR TITLE
Improve MacPorts Support

### PR DESCRIPTION
**Pull request by linuxgemini**
This change also makes:
  - ~~Setting different CC, CXX and LD values pre execution~~ Removed
  - Re-order imports so that the libraries found by the build system gets imported before general imports, avoiding version collision

This PR tracked the changes requested at https://github.com/macports/macports-ports/pull/11652

### CODE REVIEW

**Comment by doegox** [comment]
Apaprently it fails on the CI.
BTW using ?= is quite fragile if the vars are defined in one's env by default.
using = allows already override if provided as arg to make, e.g. make CC=clang

**Comment by linuxgemini** [comment]
> Apaprently it fails on the CI.
> BTW using ?= is quite fragile if the vars are defined in one's env by default.
> using = allows already override if provided as arg to make, e.g. make CC=clang

ah true, I'm removing them then

File: client/Makefile [hunk 1]
```diff
@@ -25,20 +25,21 @@ vpath %.dic dictionaries
...
+	INCLUDES += -I$(BREW_PREFIX)/include
+	LDLIBS += -L$(BREW_PREFIX)/lib
+	PKG_CONFIG_ENV := PKG_CONFIG_PATH=$(BREW_PREFIX)/opt/qt/lib/pkgconfig
+	PKG_CONFIG_ENV := PKG_CONFIG_PATH=$(BREW_PREFIX)/opt/qt5/lib/pkgconfig
 endif
 
+ifdef ($(USE_MACPORTS),1)
+	INCLUDES += -I$(MACPORTS_PREFIX)/include
+	LDLIBS += -L$(MACPORTS_PREFIX)/lib
+	PKG_CONFIG_ENV := PKG_CONFIG_PATH=$(MACPORTS_PREFIX)/lib/pkgconfig
```
Comment (doegox) [inline client/Makefile:37]: no need for the 
```
PKG_CONFIG_ENV := PKG_CONFIG_PATH=$(MACPORTS_PREFIX)/libexec/qt/lib/pkgconfig
PKG_CONFIG_ENV := PKG_CONFIG_PATH=$(MACPORTS_PREFIX)/libexec/qt5/lib/pkgconfig
```
that were there previously ?

Comment (doegox) [reply client/Makefile:37]: BTW CI still failing..

**Comment by linuxgemini** [comment]
```
gcc -MT obj/proxmark3.o -MMD -MP -MF obj/proxmark3.Td -Wall -Werror -O3 -fstrict-aliasing -pipe -Wbad-function-cast -Wredundant-decls -Wmissing-prototypes -Wchar-subscripts -Wshadow -Wundef -Wwrite-strings -Wunused -Wuninitialized -Wpointer-arith -Winline -Wformat -Wformat-security -Winit-self -Wmissing-include-dirs -Wnested-externs -Wmissing-declarations -Wempty-body -Wignored-qualifiers -Wmissing-field-initializers -Wtype-limits -Wold-style-definition -Wcast-align -Wno-error=cast-align -Wswitch-enum -Wno-error=switch-enum -Wno-strict-prototypes -I./src -I./include -I../include -I../common -I../common_fpga  -I./deps/amiitool -I./deps/cliparser -I./deps/hardnested -I./deps/liblua -I./deps/jansson -I../common/mbedtls -I./deps/reveng -I./deps/tinycbor -I./deps/whereami  -I/usr/local/Cellar/python@3.9/3.9.6/Frameworks/Python.framework/Versions/3.9/include/python3.9  -DHAVE_READLINE -DHAVE_PYTHON -DHAVE_LUA_SWIG -DHAVE_PYTHON_SWIG -c -o obj/proxmark3.o src/proxmark3.c
src/proxmark3.c:292:13: error: use of undeclared identifier 'rl_catch_signals'
            rl_catch_signals = 1;
            ^
src/proxmark3.c:293:13: error: implicit declaration of function 'rl_set_signals' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
            rl_set_signals();
            ^
src/proxmark3.c:375:21: error: use of undeclared identifier 'rl_event_hook'
                    rl_event_hook = check_comm;
                    ^
3 errors generated.
make[1]: *** [obj/proxmark3.o] Error 1```

File: client/Makefile (hunk 1 above)
Comment (linuxgemini) [reply client/Makefile:37]: yeah the macports maintainers notified me that having QT5 enabled in their Portfile will append them automatically

**Comment by linuxgemini** [comment]
looks like the ordering of the imports actually break the builds with brew

(also do note, this pr does not have to be merged, as macports do support patchfiles before building)

**Comment by doegox** [comment]
```
 ifneq ($(strip $(BREW_PREFIX)),)
    USE_BREW = 0
```
Shouldn't it be ifeq ? (if BREW_PREFIX="", then USE_BREW=0)

**Comment by linuxgemini** [comment]
> ```
>  ifneq ($(strip $(BREW_PREFIX)),)
>     USE_BREW = 0
> ```
> 
> Shouldn't it be ifeq ? (if BREW_PREFIX="", then USE_BREW=0)

oh yeah, i'll go into that

**Comment by doegox** [comment]
Could you add a line in the CHANGELOG ?
Then I'll merge, thanks

**Comment by doegox** [comment]
BTW status is draft, you still need to refine it ?

**Comment by linuxgemini** [comment]
> BTW status is draft, you still need to refine it ?

No, I'll do the changelog modification then I'll remove the draft status

**Pull request merged**